docs-src/modules/developing/pages/specFiles.adoc

```diff
@@ -0,0 +1,241 @@
+----
+====
+
+An example export result is shown below
+
+[source,yaml]
+----
+tables:
+- tableName: GamePlay
+  timeColumnName: event_at
+  entityKeyColumnName: entity_key
+  subsortColumnName: offset
+  groupingId: User
+  source:
+    kaskada: {}
+- tableName: Purchase
+  timeColumnName: event_at
+  entityKeyColumnName: entity_id
+  groupingId: User
+  source:
+    kaskada: {}
+views:
+- view_name: CountPurchase
+  expression: count(Purchase)
+materializations:
+- materializationName: PurchaseStats
+  expression: |
+    {
+        time: Purchase.purchase_time,
+        entity: Purchase.customer_id,
+        max_amount: Purchase.amount | max(),
+        min_amount: Purchase.amount | min(),
+        count: CountPurchase,
+    }
+  destination:
+    objectStore:
+      fileType: FILE_TYPE_PARQUET
+      outputPrefixUri: s3://my-bucket/materialization-output-prefix/
+  slice: {}
+----
+
+Alternately, if you know a specific table, view, or materialization you'd like to export you can specify it explicitly.
+
+[tabs]
+====
+CLI::
++
+[source,bash]
+----
+kaskada-cli sync export --table Purchase
+kaskada-cli sync export --view CountPurchase
+kaskada-cli sync export --materialization PurchaseStats
+----
+====
+
+[source,yaml]
+----
+tables:
+- tableName: Purchase
+  timeColumnName: event_at
+  entityKeyColumnName: entity_id
+  groupingId: User
+  source:
+    kaskada: {}
+
+views:
+- view_name: CountPurchase
+  expression: count(Purchase)
+
+materializations:
+- materializationName: PurchaseStats
+  expression: |
+    {
+        time: Purchase.purchase_time,
+        entity: Purchase.customer_id,
+        max_amount: Purchase.amount | max(),
+        min_amount: Purchase.amount | min(),
+        count: CountPurchase,
+    }
+  destination:
+    objectStore:
+      fileType: FILE_TYPE_PARQUET
+      outputPrefixUri: s3://my-bucket/materialization-output-prefix/
+  slice: {}
+----
+
+== Updating Kaskada to reflect the contents of a spec file
+
+To upteda a resource (table, view, or materialization), you first modify the resource in your spec file, 
+then use the `spec plan` command to preview the changes that will be made to the system. To make the 
+acutal changes, use the `spec apply` command.
+
+When a spec file is update, the CLI inspects all of the server's resources and all of the resources
+defined in your spec file, then takes whatever actions are necessary to reconcile the server's state.
+Applying a spec can create new resources, or update resources by deleting them & then recreating them.
+
+Note, if you remove a resource from a spec file, it will not be deleted from the system. Instead you 
+must delete those resources using the standard delete commands.
+
+[WARNING]
+.Table updates are destructive
+====
+Tables are currently immutable. 
+When the CLI updates a table, it does so by deleting the table and re-creating it.
+When this happens, all data previously loaded into the table is lost.
+====
+
+=== Previewing the changes
```

Comment:
Maybe move the text on line 227 here. A section followed immediately by an example comes across as a little bare.